torch/_torch_docs.py

```diff
@@ -3091,7 +3091,23 @@ def merge_dicts(*dicts):
            r"""
 flatten(input, start_dim=0, end_dim=-1) -> Tensor
 
-Flattens a contiguous range of dims in a tensor.
+Flattens :attr:`input`, or if :attr:`start_dim` or :attr:`end_dim` are specified, a range of :attr:`input`
+dimensions.
+
+Flattening a tensor reshapes it into a one-dimensional tensor while preserving its order.
+If :attr:`start_dim` or :attr:`end_dim` are specified, :attr:`end_dim` cannot come before :attr:`start_dim`,
+and only dimensions starting with :attr:`start_dim` and ending with :attr:`end_dim` are flattened.
+
+If `start_dim < end_dim` (taking into account dimension wrapping), flattening an :attr:`input` with shape
+:math:`(l^*, f^+, r^*)` produces an output shape of :math:`(l^*, \prod f^+, r^*)` where :math:`l^*` is the
+sizes of the zero or more "left" dimensions before :attr:`start_dim`, :math:`f^+` is the sizes of the one
+or more dimensions between :attr:`start_dim` and :attr:`end_dim` that will be "flattened", and :math:`r^*`
+is the sizes of the zero or more "right" dimensions after :attr:`end_dim`. It is equivalent to calling
+`torch.reshape(input, shape)` where shape is :math:`(l^*, \prod f^+, r^*)` and like :func:`torch.reshape`
+returns a view when possible. See :meth:`torch.Tensor.view` for details on when a view will be returned.
+
+If `start_dim == end_dim` then this returns the original object :attr:`input` if :attr:`input` has one or more
```

Comment:
This can be the second paragraph.

"Unlike NumPy's flatten, this does not always copy input's data. If no dimensions are flattened then input itself is returned, and if input can be viewed as the flattened shape then that view is returned. Only if the input cannot be viewed as the flattened shape is input's data copied. See :meth:`torch.Tensor.view` for details on when a view will be returned."

Note that addition documenting our discrepancy with NumPy's flatten.